selfdrive/ui/navd/route_parser.cc

```diff
@@ -0,0 +1,300 @@
+#include "selfdrive/ui/navd/route_parser.h"
+
+#include <QDebug>
+#include <QGeoPath>
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QLocale>
+#include <QUrlQuery>
+#include <math.h>
+
+#include "selfdrive/common/params.h"
+#include "selfdrive/ui/qt/maps/map_helpers.h"
+
+// TODO: this seems to create duplicate coordinates in the path when decoding the polyline, so the corresponding annotations don't match up by index
```

Comment:
You can also use `geometries=geojson` in the route request to just get a list of coordinates. Then you don't have to write hacks to match the annotations by distance instead of index.

Reply:
good idea